pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -426,10 +440,26 @@ func (p *PoolManager) AddMacToWaitingConfig(allocations map[string]string, paren
 			configMap.Data = map[string]string{}
 		}
 
-		for _, macAddress := range allocations {
-			logger.V(1).Info("add mac address to waiting config", "macAddress", macAddress)
-			macAddress = strings.Replace(macAddress, ":", "-", 5)
-			configMap.Data[macAddress] = time.Now().Format(time.RFC3339)
+		if macChanges.allocations != nil {
+			for iface, macAddress := range macChanges.allocations {
+				logger.V(1).Info("add mac address addition command to waiting config", "macAddress", macAddress)
+				macAddressDashes := strings.Replace(macAddress, ":", "-", 5)
+				configMap.Data[macAddressDashes], err = createConfigMapEntry(namespacedVmName, iface, time.Now().Format(time.RFC3339), AllocationStatusWaitingForPod)
+				if err != nil {
+					return errors.Wrapf(err, "Failed to create configmap Entry mac %s addition", macAddressDashes)
+				}
```

Comment:
Let's pass `macAddress` directly to `createConfigMapEntry` so the transformation into ":", "-" is done there I would say that also the `time.Now().Format(time.RFC3339)` can be done there.

Reply:
but macAddress is only part of the configmap's key, not value

totally agree about the TimeStamp.. Done that.